OneCommunityGlobal/HighestGoodNetworkApp
# Yubo - Read-only HOC make non-editable task view obvious

# Description
<img width="714" alt="Frontend_Task143_PR_description" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/bec062f8-9fb6-4faf-b33a-97aa3d7c4dcd">

Fixes 16 in **PROJECTS/TASKS/WBS COMPONENT** (medium)

## Related PRS (if any):
This frontend PR solves the problem the same as in #860. However, I chose a better solution for better review experience and remove unnecessary refactoring.

## Main changes explained:

1. Created a HOC(Higher Order Component) to wrap sections in `TaskEditModal` and determine Now in Task View, users who only have **view** permission (e.g. Volunteer user) can only see the contents rather than some inputs .
2. Make the fields in `EditTaskModal` in the Wrapper as much as possible, or disable the input to make the style obviously non-editable for those who only has **view** permission.
3. Add a **disable** to the props of **TagsSearch** component to disable the input.

## How to test:

- check into current branch
- do `npm install` and `npm run start:local` to run this PR locally
- You need to login as a role that **DOES NOT** have either **suggest** or **edit** permission (not volunteer). You can do this by creating a new role.

- go to dashboard→ Click any task in **Task** tab -> click **View** button to see the details of each task.
The fields that user can't edit should be obvious. Clicking any button won't make changes to the filed.

Also log as **admin** or **manager** , they have either **edit** or **suggest** permission, so the fields can be edited and has `update` button. Also try if **Add Task** button can show the Modal properly.


## Screenshots or videos of changes:
<img width="1023" alt="Screenshot 2023-12-23 at 22 27 57" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/7d1d2a09-3e73-4038-b423-e478ff0721c6">


## Note:
### 2. If you don't have an Owner account or for some reasons can't use the role, see below.
- you can modify the file `src/components/Projects/WBS/WBSDetail/EditTask/EditTaskModal.jsx` to remove permission
**It's not recommended**
<img width="487" alt="Screenshot 2023-12-23 at 22 16 08" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/35792035-a73a-4e8a-802e-516456b79de5">
